Fix mismatched Yul optimizer step lists in docs

mau211 opened the pull request:
fixes #12081,

Made updates to references reported in pr #12081

Commit:
feat(docs): get mismatched Yul optimizer step lists back in sync

cameel commented on docs/internals/optimizer.rst:
```diff
@@ -1109,7 +1109,7 @@ This component can only be used on sources with unique names.
 
-FullFunctionInliner
+FullInliner
 ^^^^^^^^^^^^^^^^^^^
```
```suggestion
FullInliner
^^^^^^^^^^^
```

mau211 replied:
I committed your suggestion via GitHub

cameel commented on docs/yul.rst:
```diff
@@ -1263,6 +1263,7 @@ Abbreviation Full name
 ``a``        ``SSATransform``
 ``t``        ``StructuralSimplifier``
 ``u``        ``UnusedPruner``
+``p``        ``UnusedFunctionParamterPruner``
```
```suggestion
``p``        ``UnusedFunctionParameterPruner``
```

cameel commented on docs/internals/optimizer.rst:
```diff
@@ -1084,7 +1084,7 @@ Function Inlining
 
 .. _functional-inliner:
```
The references need to be updated too.

cameel commented on docs/internals/optimizer.rst:
```diff
@@ -1084,7 +1084,7 @@ Function Inlining
 .. _functional-inliner:
 
-FunctionalInliner
+ExpressionInliner
```
There's a section called `Optimizer Steps` above. It has a list of these steps and it still has the old names.

mau211 replied:
Noted, updated a field for Optimizer steps section

cameel reviewed (commented):
Since this PR fixes #12081, you should add `Fixes #12081` to the PR description. This way github will close the issue automatically when this is merged.

Also, there's no need to copy the whole issue description to the PR. If you really want to quote it, please use proper syntax, so that it does not look like your own text. I.e.

```markdown
> Looking at the list of Yul transformations in the documentation, I noticed a few slight differences between the list at https://docs.soliditylang.org/en/latest/yul.html#optimization-step-sequence and the list at https://docs.soliditylang.org/en/latest/internals/optimizer.html#optimizer-steps:
>
> - [ ] In the second list, change FunctionalInliner to ExpressionInliner, just like in the first list.
> - [ ] In the second list, change FullFunctionInliner to FullInliner, just like in the first list.
> - [ ] Add UnusedFunctionParamterPruner to the first list.
>
> Reported by @acoglio 
```

will be rendered like this:

> Looking at the list of Yul transformations in the documentation, I noticed a few slight differences between the list at https://docs.soliditylang.org/en/latest/yul.html#optimization-step-sequence and the list at https://docs.soliditylang.org/en/latest/internals/optimizer.html#optimizer-steps:
>
> - [ ] In the second list, change FunctionalInliner to ExpressionInliner, just like in the first list.
> - [ ] In the second list, change FullFunctionInliner to FullInliner, just like in the first list.
> - [ ] Add UnusedFunctionParamterPruner to the first list.
>
> Reported by @acoglio

cameel reviewed (approved):
A link to the Full Inliner was still missing and it was still called "Full Function Inliner" in two places in the text but I just pushed a change to fix those. I also squashed your commits (we usually do one commit per logical change and do not leave review fixes as separate commits) and rebased it on `develop`.

In this form the PR seems done do me. Thanks for submitting it!

mau211 commented:
> A link to the Full Inliner was still missing and it was still called "Full Function Inliner" in two places in the text but I just pushed a change to fix those. I also squashed your commits (we usually do one commit per logical change and do not leave review fixes as separate commits) and rebased it on `develop`.
> 
> In this form the PR seems done do me. Thanks for submitting it!

@cameel thank you very much :)

mau211 commented:
@cameel I do not have access to merge it. Can you merge the changes and close the pr.
